database/misc/mongodb_parameter.py

```diff
@@ -0,0 +1,244 @@
+author: "Loic Blot (@nerzhul)"
+'''
+
+EXAMPLES = '''
+# Set MongoDB syncdelay to 60 (this is an int)
+- mongodb_parameter: param="syncdelay" value=60 param_type="int"
+'''
+
+RETURN = '''
+changed:
+    description: If module has changed MongoDB parameter
+    returned: success
+    type: string
+before:
+    description: value before modification
+    returned: success
+    type: string
+after:
+    description: value after modification
+    returned: success
+    type: string
+'''
+
+import ConfigParser
+
+try:
+    from pymongo.errors import ConnectionFailure
+    from pymongo.errors import OperationFailure
+    from pymongo import version as PyMongoVersion
+    from pymongo import MongoClient
+except ImportError:
+    try:  # for older PyMongo 2.2
+        from pymongo import Connection as MongoClient
+    except ImportError:
+        pymongo_found = False
+    else:
+        pymongo_found = True
+else:
+    pymongo_found = True
+
+
+# =========================================
+# MongoDB module specific support methods.
+#
+
+def load_mongocnf():
+    config = ConfigParser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
+        creds = dict(
+            user=config.get('client', 'user'),
+            password=config.get('client', 'pass')
+        )
+    except (ConfigParser.NoOptionError, IOError):
+        return False
+
+    return creds
+
+
+# =========================================
+# Module execution.
+#
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(default=None),
+            login_password=dict(default=None),
+            login_host=dict(default='localhost'),
+            login_port=dict(default='27017'),
+            login_database=dict(default=None),
+            replica_set=dict(default=None),
+            param=dict(default=None),
+            value=dict(default=None),
+            param_type=dict(default="str"),
+            ssl=dict(default=False),
+        )
+    )
+
+    if not pymongo_found:
+        module.fail_json(msg='the python pymongo module is required')
+
+    valid_param_types = ['str', 'int']
+
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    login_host = module.params['login_host']
+    login_port = module.params['login_port']
+    login_database = module.params['login_database']
+
+    replica_set = module.params['replica_set']
+    ssl = module.params['ssl']
+
+    param = module.params['param']
+    param_type = module.params['param_type']
+    value = module.params['value']
+
+    if param is None:
+        module.fail_json(msg='Cannot run without param option')
+
+    if value is None:
+        module.fail_json(msg='Cannot run without value option')
+
+    if param_type not in valid_param_types:
+        module.fail_json(msg='param_type should have one of the following value: %s' % str(valid_param_types))
```

Comment:
This check can be removed after `choices=['str', 'int']`has been added, see comment above